src/EscrowERC20.sol

```diff
@@ -53,6 +54,18 @@ contract EscrowERC20 {
         playerToContractToAmount[playerAddress][contractAddress] = _Amount;
     }
 
+    function transferTokens(address _PlayerAddress, address _ContractAddress, uint256 _Amount)
+        public
+        inEscrAvailable(EscrERC20Available.YES)
+        onlyEscrow
+    {
+        playerAddress = _PlayerAddress;
+        contractAddress = _ContractAddress;
+        bool success = ERC20(contractAddress).transfer(playerAddress, _Amount);
+        if (!success) revert EscrowERC20__TransferFailed(address(this), playerAddress, _Amount);
+        totalBalance -= _Amount;
```

Comment:
It's a good practice to do all the state changes before any external contract calls to prevent re-entrancy. Please transfer the tokens at last.